sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1332,4 +1430,36 @@ public synchronized Throwable fillInStackTrace() {
     }
 
     // endregion
+
+    // TODO: remove when we are confident of RNTBD OOM bug
+    @SuppressWarnings("unchecked")
+    private ScheduledFuture<?> startMonitoring() {
+        return monitoringRntbdChannelPool.scheduleAtFixedRate(() -> {
+            int i = getTaskCount();
+            if (isInterestingEndpoint()) {
+                logger.debug("{} total number of tasks on the executor [{}], connecting [{}], acquiredChannel [{}], availableChannel [{}], pending acquisition [{}]",
+                    this.hashCode(), i, connecting.get(), acquiredChannels.size(), availableChannels.size(), pendingAcquisitions.size());
+            }
+        }, 0, 60, TimeUnit.SECONDS);
+    }
```

Comment:
One more I found useful is to list the channels and their interesting states (specifically the pendingoperations on channel was super useful). 
It can be filtered on some threshold.

Reply:
ACK - But i think we can decide that when we want to remove this additional logging. IMO for now we should leave it in here until we have confirmation that the hotfix addressed all scenarios.